### DISCUSSION TITLE
append an element if total value of usefield array is 100

### DISCUSSION THREAD

**Discussion by Sbabureddy** [react-hook-form/react-hook-form#11577] [General]
Hello everyone, I'm working on useField array, I've a requirement to append an element if total value of usefield array is less than or equal to 100. while appending the element useEffect is firing infinite times. can anyone help me to achieve this. thanks in advance.

https://codesandbox.io/p/sandbox/watch-subscription-xgzxmj

**Comment by Moshyfawn** [comment]
Your business logic is impossible and causes infinite re-rendering. If you get a sum of 100, you will append fields infinitely because the value will always be 100.

**Reply by Moshyfawn** [reply]
I may have misread your business logic, but my comment still applies: appending fields with a value of 0 will always cause re-renders, because the sum will always remain 0.

**Reply by Sbabureddy** [reply]
@Moshyfawn thanks for the quick reply.
The requirement is user has to enter the  value which total value of useField array is  equal to 100. If the value is less than 100. An input has to automatically popup. Can you suggest solution for this?
